spec/models/efile_submission_spec.rb

```diff
@@ -363,13 +363,31 @@
       allow(Irs8812Ty2021Pdf).to receive(:new).and_return(instance_double(Irs8812Ty2021Pdf, output_file: example_pdf))
     end
 
-    it "generates and stores the 1040 and 8812 combined PDF" do
-      expect { submission.generate_filing_pdf }.to change(Document, :count).by(1)
-      doc = submission.client.documents.last
-      expect(doc.display_name).to eq("IRS 1040 - TY #{TaxReturn.current_tax_year} - #{submission.irs_submission_id}.pdf")
-      expect(doc.document_type).to eq(DocumentTypes::Form1040.key)
-      expect(doc.tax_return).to eq(submission.tax_return)
-      expect(doc.upload.blob.download).not_to be_nil
+    context "the filer is claiming CTC (line 28 is greater than $0)" do
+      before do
+        create(:qualifying_child, intake: submission.intake)
+      end
+
+      it "generates and stores the 1040 and 8812 combined PDF" do
+        expect { submission.generate_filing_pdf }.to change(Document, :count).by(1)
+        doc = submission.client.documents.last
+        expect(doc.display_name).to eq("IRS 1040 - TY #{TaxReturn.current_tax_year} - #{submission.irs_submission_id}.pdf")
+        expect(doc.document_type).to eq(DocumentTypes::Form1040.key)
+        expect(doc.tax_return).to eq(submission.tax_return)
+        expect(doc.upload.blob.download).not_to be_nil
+      end
+    end
+
+    context "the filer is not claiming CTC (line 28 = $0 or missing)" do
+      it "generates and stores just the 1040 (does not generate the 8812)" do
+        expect { submission.generate_filing_pdf }.to change(Document, :count).by(1)
+        expect(Irs8812Ty2021Pdf).not_to have_received(:new)
```

Comment:
yeah i wonder if there is a way to count the number of pages in the doc as well but this is probably sufficient as a check